daml-lf/spec/daml-lf-1.rst

```diff
@@ -565,12 +560,24 @@ strings as *package identifiers*.  ::
   Package versions
            pversion ::= PackageVersionString        -- PackageVersion
 
-We do not specify an explicit syntax for contract identifiers as it is
-not possible to refer to them statically within a program. In
-practice, contract identifiers can be created dynamically through
-interactions with the underlying ledger. See the `operation semantics
-of update statements <Update Interpretation_>`_ for the formal
-specification of those interactions.
+  V0 Contract identifiers:
+          cidV0  ∈  #[a-zA-Z0-9\._:-#/ ]+           -- V0ContractId
+
+  V1 Contract identifiers:
+          cidV1  ∈  00([0-9a-f][0-9a-f]){32,190}    -- V1ContractId
+
+  Contract identifiers:
+          cid := cidV0 | cidV1                      -- ContractId
+
+Contract identifiers can be created dynamically through interactions
+with the underlying ledger. See the `operation semantics of update
+statements <Update Interpretation_>`_ for the formal specification of
+those interactions. Contract identifiers are naturally ordered
+lexicographically. DAML-LF >= 1.dev programs creates only V1 Contract
```

Comment:
What is a DAML-LF 1.dev program? Most programs (the default output of any `daml build`) consist of multiple LF versions. I assume what you mean is that a contract id of a template in 1.dev will always use the new format?

Reply:
Good point. Let me reformulate that.